File: practice-2/rust-exercices/src/load_keypair.rs
use solana_sdk::signature::Keypair as SolanaKeypair;
use std::env;

pub fn load_keypair() -> SolanaKeypair {
    dotenv::dotenv().ok();

    let secret_key_string = env::var("SECRET_KEY").expect("SECRET_KEY not set");

    let secret_key_bytes: Vec<u8> =
        serde_json::from_str(&secret_key_string).expect("Failed to parse SECRET_KEY as byte array");

    if secret_key_bytes.len() != 64 {
        panic!(
            "Invalid secret key length: expected 64 bytes, got {}",
            secret_key_bytes.len()
        );
    }

    SolanaKeypair::from_bytes(&secret_key_bytes).expect("Failed to create Solana Keypair")
}

File: practice-3/recipient-pays-for-transaction/load-keypair.ts
import "dotenv/config";
import { Keypair } from "@solana/web3.js";

export const load_keypair = (env_name: string): Keypair => {
  const asArray = Uint8Array.from(JSON.parse(process.env[env_name]!));
  return Keypair.fromSecretKey(asArray);
};
